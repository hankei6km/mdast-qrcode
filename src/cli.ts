import { Readable, Writable } from 'stream';
import { fromMarkdown } from 'mdast-util-from-markdown';
import { toMarkdown } from 'mdast-util-to-markdown';
import { toImageDataURL } from './qrcode.js';
import { frontmatter } from 'micromark-extension-frontmatter';
import {
  frontmatterFromMarkdown,
  frontmatterToMarkdown
} from 'mdast-util-frontmatter';

type ToMarkdownOptions = {
  bullet?: '-' | '*' | '+';
  closeAtx?: boolean;
  emphasis?: '_' | '*';
  fence?: '~' | '`';
  fences?: boolean;
  incrementListMarker?: boolean;
  listItemIndent?: 'tab' | 'one' | 'mixed';
  quote?: '"' | "'";
  resourceLink?: boolean;
  rule?: '-' | '_' | '*';
  ruleRepetition?: number;
  ruleSpaces?: boolean;
  setext?: boolean;
  strong?: '_' | '*';
  tightDefinitions?: boolean;
};

type Opts = {
  stdin: Readable;
  stdout: Writable;
  stderr: Writable;
  bullet?: ToMarkdownOptions['bullet'];
  rule?: ToMarkdownOptions['rule'];
};

const cli = async ({
  stdin,
  stdout,
  stderr,
  bullet = '-',
  rule = '-'
}: Opts): Promise<number> => {
  try {
    let source = '';
    await new Promise((resolve) => {
      stdin.on('data', (d) => (source = source + d));
      stdin.on('end', () => resolve(source));
    });
    const tree = fromMarkdown(source, {
      extensions: [frontmatter(['yaml', 'toml'])],
      mdastExtensions: [frontmatterFromMarkdown(['yaml', 'toml'])]
    });
    await toImageDataURL(tree);
    stdout.write(
      toMarkdown(tree, {
        bullet,
        rule,
        extensions: [frontmatterToMarkdown(['yaml', 'toml'])]
      })
    );
  } catch (err: any) {
    stderr.write(err.toString());
    stderr.write('\n');
    return 1;
  }
  return 0;
};

export default cli;
