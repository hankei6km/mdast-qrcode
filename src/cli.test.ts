import { PassThrough } from 'stream';
import cli from './cli';

describe('cli()', () => {
  it('should convert "qrcode:" to DataURL', async () => {
    const stdin = new PassThrough();
    const stdout = new PassThrough();
    const stderr = new PassThrough();
    let outData = '';
    stdout.on('data', (d) => (outData = outData + d));
    let errData = '';
    stderr.on('data', (d) => (errData = errData + d));
    process.nextTick(() => {
      stdin.write('# test1\n\n![](qrcode:test1)\ntest1');
      stdin.end();
    });
    expect(
      await cli({
        stdin,
        stdout,
        stderr
      })
    ).toEqual(0);
    expect(outData).toEqual(
      '# test1\n\n![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHQAAAB0CAYAAABUmhYnAAAAAklEQVR4AewaftIAAAKmSURBVO3BQW7EVgwFwX6E7n/ljnfh6gOCNBObYVX8wRqjWKMUa5RijVKsUYo1SrFGKdYoxRqlWKMUa5RijVKsUYo1SrFGKdYoFw8l4ZtUuiR0KidJ6FROkvBNKk8Ua5RijVKsUS5epvKmJJyonCShU+mS0KmcqLwpCW8q1ijFGqVYo1x8WBLuUPmkJLwpCXeofFKxRinWKMUa5eKPS8L6V7FGKdYoxRrlYhiVO5IwSbFGKdYoxRrl4sNUfpMkdCpPqPwmxRqlWKMUa5SLlyXhN0lCp9IloVM5ScJvVqxRijVKsUa5eEjlL0nCHSp/SbFGKdYoxRrl4qEkdCp3JKFT6ZJwRxLuUOmS8CaVkyR0Kk8Ua5RijVKsUeIPXpSETqVLwonKHUnoVLok3KFykoROpUvCHSpvKtYoxRqlWKNcfFgSTlS6JJyonCThDpWTJJwkoVM5ScInFWuUYo1SrFHiD74oCXeoPJGETuUkCScqXRI6lf9SsUYp1ijFGuXioSR8UhI+KQlPqHRJ6FROktCpPFGsUYo1SrFGuXhI5b+UhBOVb1LpktCpdCpvKtYoxRqlWKNcPJSEb1I5UemS0KncofKmJHQqbyrWKMUapVijXLxM5U1JOFHpknCShBOVkyR0KicqXRI+qVijFGuUYo1y8WFJuEPljiR0KidJ6FROknCShBOVTqVLQqfyRLFGKdYoxRrl4o9T6ZLQqZwkoVPpVO5IQpeEbyrWKMUapVijXPzPqZwk4QmVbyrWKMUapVijXHyYyjepdEnoVE6ScKLSJeE3KdYoxRqlWKNcvCwJ35SETqVTOUnCEypdEu5IQqfyRLFGKdYoxRol/mCNUaxRijVKsUYp1ijFGqVYoxRrlGKNUqxRijVKsUYp1ijFGqVYoxRrlH8AuV30+O6ObzcAAAAASUVORK5CYII=)\ntest1\n'
    );
    expect(errData).toEqual('');
  });
});
