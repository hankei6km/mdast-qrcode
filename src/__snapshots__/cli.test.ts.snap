// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli() should convert "qrcode:" to DataURL 1`] = `
"---
title: title1
---

# test1

![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHQAAAB0CAYAAABUmhYnAAAABmJLR0QA/wD/AP+gvaeTAAACQElEQVR4nO3dwW6kQAwA0WWV//9lcu8DjmW7aUr1rkmYjEoti2lgrvu+73/C+P/2P6BeBoUxKIxBYQwKY1AYg8IYFMagMAaFMSiMQWEMCmNQGIPC/FQPcF1Xx//xZ+v27fr60fZu9fenVbenXaEwBoUxKEx5hq66L1GKZlh1ZmZn6u73l+UKhTEojEFh2mfoKjsjpi8T7p5Zp70/VyiMQWEMCjM+Q6ft/qz1dK5QGIPCGBTm8zN0lT3Po81gVyiMQWEMCjM+Q097hEN2/zNy2vtzhcIYFMagMO0z9LTzuu5rik57fytXKIxBYQwKU56hp52HRU67BqibKxTGoDAGhWm/P7S6HxmdJ2aPF6m+Xvb4q+7Pll2hMAaFMSjM1f3M+epMfHsGd99vWn29LFcojEFhDArz+v2h3TN2emZGP397P9UVCmNQGIPCtJ+Hhi+4+Tyte8ZVr0ma5gqFMSiMQWGOf17u6cff/fzeiCsUxqAwBoX5/HW53furVdmZ6n6oHhkUxqAwx5+HrrLPhK+e500/Y94ZqkcGhTEoDO57W7LXAEXHi/6+OtO7uUJhDApjUJjXr8tddd/b0v372Z9Hn926H6pHBoUxKMznnzk/fU1Pdabv/qzbFQpjUBiDwnx+hlZN38+5+5omVyiMQWEMCoP73pbpZ8rv3t/McoXCGBTGoDCf/96W6mex0/eHRtwP1SODwhgUZvtzijTLFQpjUBiDwhgUxqAwBoUxKIxBYQwKY1AYg8IYFMagMAaFMSjML7ld9PiTOS1MAAAAAElFTkSuQmCC)
test1

---

# page2

-   item1
-   item2
"
`;

exports[`cli() should convert "qrcode:" with logo to DataURL 1`] = `
"---
title: title1
---

# test1

![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASwAAAEsCAYAAAB5fY51AAAABmJLR0QA/wD/AP+gvaeTAAAgAElEQVR4nO3dWXMcx7Ut4JWZVT2jgcZMAuAsirLkQWeI+3TvXxP5s+6b4z4cx4k4x3EsyZbEQSRBAsTcaDR6rsy8D7saM0hQBglk9focsGUK6G5Mi7t27cxU3nsPIqIA6Kt+AUREF8XAIqJgMLCIKBgMLCIKBgOLiILBwCKiYDCwiCgYDCwiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiILBwCKiYDCwiCgYDCwiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiILBwCKiYDCwiCgYDCwiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiIIRXfULuCil1FW/hGvNe3/h973sr+XHPPdV4s/Q+4XwfWSFRUTBYGARUTAYWEQUDAYWEQWDgUVEwWBgEVEwGFhEFAwGFhEFg4FFRMEIZtL9okKY1v0Yn2I6O2tfo8uWta9Plib8WWERUTAYWEQUDAYWEQWDgUVEwWBgEVEwGFhEFAwGFhEFg4FFRMFgYBFRMBhYRBSMzC3N+RhXuWQhS8s/RnX5EH9+Pj9WWEQUDAYWEQWDgUVEwWBgEVEwGFhEFAwGFhEFg4FFRMFgYBFRMBhYRBSMkZ50H1UXndAe1Wlqur5YYRFRMBhYRBQMBhYRBYOBRUTBYGARUTAYWEQUDAYWEQWDgUVEwWBgEVEwOOk+gq5ygv2iz81pfDoLKywiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiIIx0ktzuKzj/S66PGZU8efn82OFRUTBYGARUTBG+pIwBI8fP+ZzX+PHpM9L+UAuxLndCNFvk6XfHV4SElEwGFhEFAwGFhEFg4FFRMFgYBFRMDIXWEqp4N6ePHly1V82+oAnT55c+c/Jb33LkswFFhFlFwOLiILBwCKiYHBpDl2Kgynp4bB02jrJWg+FrhYDiy6P94B18s9GAwwrumQMLDrTsYrJOcBaeGslkKyTP3NeQsp7eX/rgCSRj4siwGipsI6+aQVoLf/OGCAy8v+1hBsrMnofBhadz3sJqnYPbrcJv7cP32zD77fh21347gAYDIDEAkkCP7DAYBhYEkYqMhJekYHKx1CFHFAuQlVK0NUKVG0MqlKEysWAMVf7+dK1x8AiAEcqKicVlO8P4HsDoNOF227ArW7CvtuC29iB29qFr+/BNzvw3S7Q7cP3+hJWiZXH0RqIDZDLSVDlc1DlAlS1AlWrQs/WoOenYRZnoWdqUNUyVLEAn4sl5NJLSlZcdBQDiw5ZC98bwNebcOvbcOs7ElDDt61duPoefGM/rbJ68P0+0E8OKy2XBp+CVExxBMQRVCzVlSoXocZKUBNj0FPjsDM1CazZGvTcFPT8NPTUOFSlBJXPHTTviYCA9sOiS5Z+2z0gvaf+AL7dhavvwb5+h+TnV7BPl2FfrsBt1YF2T6qo4aXfsJ/lnDzWsJ91lAKg9OneVXQkyHIx1MQYzMIM9N0FRF/dhbm3CHNjGqpWlUvIyBw08FlxjTZWWKPOOvhmSyqotxuwv64gebkiQfVmHW5tC77ZlnD62L/bPACf3jW0x//42D9v1uG2GzCbu3AbOzDLa4ju3oS+fRPm1jzUZBUqjiTwaKSxwhoxh3f/PDBI4Ntd2JUN2H+8RPLDMyQ/PId9u3HQWEevL9XUyZ+SYfVk0jt8w7uAx54s/S/n07uK7vCS8Sitpdoq5IBiHnqmBnNvEdHX9xD/61cwD5agJsagCnl5X8VKa1SxwhpF3sP3E/itOuzLFbn8+/EFkl9ew75cga83Dy/1zqKU3PUr5qHGSkCxIBVQHEnfSkE+PpExB9/tw7c68B1p0B805oecA3pp476xD7vXgt/vHIRmtL0H8+g2zM1ZoJw+F40kfudHxNHKyvcG8LtN2KfLGPzleyTfP0t7VbtSVbkTFZWCVDbpiILKx1ClAnStCjU3BV0bA4p5qEJeGuVawScW6A3gOz345r489s4e3G4TaHfl3w/fnDv+Ynt9uLUteb+NOtzKJuJuD/CAvjENNVaCT/tarLRGCwNrlHgP3x/ArW/D/vIKyV9/weC/f4J98QZua1eqn7PEsdy1m6rCzE1BTU9AT1ahJ8ehJsdlJCGfA3IxVC6SS0Vr4QeJVE6tDlxjH76+B7ezB7/dgNusy13H7V2g1Tl+qWgd0OnB9RO4Vge+P5Cw7PcR/eEL4PZNudMYx7yLOGIYWFl3tLLqy8iCffoa/f/3VyT/8xTu9Tu4+p6MJpxkDJCPoScq0DdmYB4sIvr6PszdBejZSZmpyscSJkYDSkPpwwTxw55VGl6+3YPf25eZrqevkTxdhn22LI39Tk9ew9Fqy1mg04Vd3YQfDGScwnnExkAvzkGNV+CVDJuy0hoNDKxRkVj4rV0kv7zG4K8/I/nbM9gXb+Eb+0B/cPr9jYGqjcEszcHcvgl9bwHm/iKi+4vQN2egJ8agivmzm+3n8IMEvtOFm5uCmhqHnptCsjAD++tb2Jer8Os7ckk6nJb3kGqr1YbrD5B4QJULgFKIAajIQJWLEpg0EhhYGSdzVha+1YF9uYrBX76Xy8BXqxJWyYnKSim5Y1cpwdxdQPy/vkH0zQOYB4tSVZWL6WxU9PGLm42GKhag5yPoahnu1jzMN/dhn77G4D/+huTHF3Br2/B7LeltHUzfe2AwgNusY/DXn+H7A+hiHqpchDYaqlQ4HHlgpZVpDKyssw5+rwX7Zh3JT79Kg/35G/j63unKSisgF0PPTsLcX0L0+weI/+13MA9vQd+Yhq6U/qmXorQGtJa7fKUC1HgFempCqiSloCplJD8+h339Dn53H+j1Dpv/zgPtrlxOxhEGMzWglEdktNw9zGn2s0YAAyujhncFfX8Au76N5MfnB3cD/Xbj8LJrSCkgn4OuVRE9uoP4//wLot8/gLl1A2pqXJrqly2tjszinFRLs5NAqQBEBvbXFfgtWVR9GFoO6A3gNupI/uepfPxYGXpi7KCPxszKtpEOrMs+wvu3Nn4fP36M77777jd97Lk84JMEfr8Nt7yO5PtnSH5+Bb+1K8OgJ8UGeqaG6Ku7iP7td4j+9CXM/UXo8cqnCSscVlwYK8GUClCRgR8k0rj3gLVWLlt7RypB5+D327DL74BSQfprM5MwSkGNla9tP+sqj4vP0lH1Ix1YmeadXEJt1GFfvJX+0MtVaWqfFEdQY2WYewuI//e3iL59BHPnBvR4RXpVn5pSUh1NjiP+05dQhbyMMvT6sHYVPmkebgwIyDDq7j7c23UkP72U8Yp8DqaQhzc6fUjWWlnEwMqqxMLtNGCfLSN5tgy3sinVysm/RbWGGq/A3L2J6JsH0mC/vwg9nK36DA42+SvkoWZjGGsR1/dk8LTXh+30gG7vcFbLeVmsvbMH+3QZSW0Mem4SerYmc2Bcc5hZDKyMOdq7cu+2kPzwDPbpa/j91tlLbXIR9PwU4m8fIfr2S5ilOehqWZbZfG7pRL2eqCL640MAPh0ubchup/3BsQl83+7CvlqBGivCfHEL5u6ChF7u4qMWFBYGVtYc9K46cKtbsM/ewL5dl8HMk3IRVLUss1bfPED05Z1P12C/AKm0IAugb0zDtNqIHt2B29qVftZ24/il4SCR6fmVDbjlNdh3mzCFHNTEGKusjGJgZY13QLcHt9OAe7cF93YdfqtxepJdK6hSUXb9vLuA6OEt6IVZ2RHhqhkNlc9BT9cQfX0PvtmCb7ZgGy3AHrlh4D0wsDK28fod9Iu3MioxVrqaCpE+Of41lDXWwTdacCsbsKsbcPVm2v85scB4OMl+fwnmi1uy2+dYSTbXu2JKKZlir5ZlM79Hd6Hnp6BKeVkCdJT38K0u7Js1mS/bbsjCasokBlZWDE+vSRK47V3Y529h36zDd8+4FASAyEDPTiL65j7MF7egKsXP+3ovQBXy0AuziO4twizMyKVePneqP+W7fbiVTdhfZccJDBL44Uk+lCkMrKxJLNzOHuyrVbh3W/Bn7cCQDmzq2RrM/QXoxTmgWPiodYGfg4qMVH2zNejFeQmtsdLBkWAHBgO4nT24tS1p0jfbMhbBwMocBlZGeKR3CBMLv9uURvTGzunlN0oB+RiqWpGByxvpoQ+5a9rzUUqm4W/Nw9xbgp6onj4OzMl2NL6xL5eEO3tAu3f27qYUtGv6U3rap5jWvewJ9o957ksfbPSAT6ycF7jblA3z9lqnFzdrJQuQp8fltJrxMejiNWi0n2V48EQhJ2sZl+ahfn51Rh8L6fmJXbjtXbjNOvT0uOzsgKvvyQGXP0U+qoOxrLCywss2w26vBddowjdbZzfbtZZDTOenoWcmZD+r6y6Xg56pyW6jldK5Iwt+kMgZimtb8vlbd+b7UbgYWFnhvGxHvNuE392X3pV1pw+P0EoWGs9NQU1NALn42vWuTlK5SA5fnalBVUuy28NZr3eQpNVlHW6/IxsIUqYwsLLCOdmKeLsBd9al4JDWUJUi9Fwt7V0FUGEZab7r2pgsGSrmz17knMhiabfdgG91T1eXFDwGVlY4JxVWXS4Hz51FSpvYemYSemo8iAFLpZUcuFouQI2PQY1XZLzhxN1Cn1g5aafeBDpdNt0ziIGVFc4D3b6EVat7bv9G6TSwpifkZOUQKqyhOJZLw+ma7DKqTvz4pjur+mZLjgxjhZU5DKyM8N7D9/rw+22pLs5rOGsN5HNy0k25eHpE4DpKe2wqiqArJajximzTfHIeyzo56GK/LXtocQ4rcxhYWeEOT3L23fdUF1pBxdGxMwSDYTRQyEGV8mnj/cS/dw6+25OvQX8gX5N0BQBlAwMrK7xPT6XpHf6ynqTlxGakpzQro8Oa59EKKp+TBdqRwanESg+r8N1+urWyP32XlILGwMqMdMq9P5D92v0ZFZYx0rzOSVhd51GGM2klE/n5nFzKnnz53gOJk0ozcfDOQ/5DWXH9bxGlPvW+6p/7uS99Ya6HXAYmFrD29GVQug0xcjEQX//Zq7MoLa//4PDWs16/99K/szadQ/Of7PMMqjrNCFZYWeLllGW4MwZGgYPAUnGg2winZyYiF0NF73v9R0LL8bIwSwL8qaVzDRvMZ/WvFNKj5NOTakIsDob7ZEVGRhrOq3BkJXga3EyrLGFgZU54l3oXNvy0hlspf+h9s/p1GGEMrCxRSu4EntnbAeCdTMCf1eMKgQe8c/DWAQ7nX+oN+3PnfS0oWAysLNH64ATk07+ochcR/cH5Yw/XnZM93DEYwDuLcxNLQe4iGpOG1ud8kfQpMbCyYngXMI7OvoPmIZPgw7GHYX8noErL+3RkoZ+cvRMFkIZ29J7gppAxsLIirSpULpbQOusX1TmpsAYJvA0rrADIFjqDAdDvy2XtycQajj0UclBRJOsmlWKBlSEMrKxQ6ZKbYh4qlzt7bMGlYw+DJA0uG9ZBDd4B/QS+N5DL27P2+irEsv1MLjq9OJqCx+9oVqjhFixFqGIOMOfUFW64hKcrOxqEtCun8/D9vuykmpxx48BoWSNZLkqlOWy687IwMxhYGaG0Agqx7GZQKpy/C4NPt6Fp7Ms2wiGd4ZcMt49pS5V1KrAMVKWY7uaQZ1BlEAPrApRSF3r7rY/55MmTS3iRaXUxXoEql2St4Bm883JYw2YdfmcPfnDOzqTXkB8k8LvpjqKd7qnAGh6+qiarQCn/yaf5h2cfXsXbp3idIWBgZUV6Go6aGJOz+847wXm4lfJmHW6ncfoYsEvgndyN9P2B/PM/+cvgvYe3FugN4Br7cPUmfKd/ejQjMnJ82eS47PUV4vIjei9+R7NCa6BUgJoah6qW3xNYXgJrfUcqlcHlBxaGx421u5dzbLxz8P0Ert2F32vJpexZB6XGkRxbNjUBVSpCndfHo2AFs1sDfYDWcodwfEyqrGJBQsueWE/nHPx+B259W451P9oL+id6Pn4402UtXF1OnvbdPnStCj2RvqbhZdrHXkIP92rfbcI320B3ONZwnIpj6Mkq1OxkekI0/z7OGgZWVmgF5GI5XSYNCJQKQLt7vLHuPPx+G3ZtC2azLpvdXdIWLD6xQLsL+3IV/f/7F7j1bej5aZgHS4j+9BBmaV62Nj6v+jvvcfsJ/PYu3Po2fKudTrmfoCAnWk9PwNychq6Ww9j+mT4KAytDVGSAglRZerYGvVqFGyTHA8un5xduN+A2d+Dqe3CtjjTsf8sJOsPqzEMuNZfXkHz/DIP/+jvcmw3o2Rrcxg6QJPD7bZj5aQnT9KiuC1VavR7cuy24N2vwjf3ToxhG9qnXExU5XGNyXMI6pO2f6UIYWBkx/MX3kYGeqMAszsGtbsllVKd3+I4+3fu92YbbqMO9WYcbnqr8W4/8Si8H/dYuBv/5AwZ/+V4ed2sXvtWW06jXtxE9W0b0r18h+vKOPF+1fKHKzrd7sK/fIXn+Bm5n73RgxRHUZFVOs54chyoV0i1oGFhZw8DKGBUZqMkqzJ2bsG/WYZfXgMb+8XdKD2twG3Ukz5ehpiegxytSlQAf9YvugYMmu1vdRPLDcyT/+BWuvgd0e3KST6sDv3U4RuFbXZgvbsEszBzOTKV7dB2tuLyTtYOusS+fy+t3Z1ZYKp+DmZuCvnVDGu6Fcyb9KXgMrKyJDPTUBMy9RZhXq0h+eAavcHoZS2LhN3eQ/PAcujaO6PYNYHL8N11G+U4P9t0W7MsV2Ffv4NZ3pDE+3Eivn8DbtgROtwf3Zh3mq7uIvr6H6JsH0Aszspzo5OzYIJHqbGMHbmUTbn0HvtU5tV+9KuRhluYRfbEENTMh6whZXWUSAytrtIaqlqEXZqAXZqGnJuA260Cnd7wysQ5udx94uQJ7cwb263tQExWosfJHH67qh1szawVdLUNPjMErBd/uHt6lTCz8bhO22YLbacDt7sHv7QODBKbVgZ6dlEvEvDTlFSQI3dt12Bdv4VY3pbo6OjeWbpmsJirQt+Zh7i5A16rAe7dPppBlLrCucmL3WvytPjzWfbwCc2Ma5vYNaay/25I7hkPeyyXbVgP25SqSv7+AKuZhHix9dGCpQh56fgrm6/uI+wlUrYrk+2dwb9cltIbT9AfBtQ/7YkXGK9a2ET16i+hfHsF8cQt6pgZVlktTv9tE8uMLJP/9E+y7zcPHGUon2/XcFMyteejFWQk9/XnuDl7lgScf46KvM4Rp98wFFkF+kUsF6BvTMF/ehqvvydrBTu/4TFZi4Vtt2JUNJN8/k0n5MalyVC6WxvUFqFwMNVGV5UFRBFUpQeVjJNUy3NoWfL0J3+4A/TRw+gP47QZsqwO3uSsDrH3pdZl7i9BTVUApabT//QWSn1/BbzdOzV6pQk4+x/uL0EtzMoNVyPPuYIYxsDLm4G/TXAx9YxrRH76A39mT/lG9CbgT27IkFm5jB8kPz+RjpieAQg56ZvJCgaXS/dW90lCVosxalQpS9Tx6g+SHZ7C/vIZ9uQJf3ztcTuOcVHiJhXUOvt2BXV5D9PV96JszUEbDrmzAPl2W0Gt3Ty3FUZUiooe3Ef3xIczCrByyyrDKNAZWVhkDVavC3FuAW9mE+eUVfLMFv3uiD+Q8sN+GfbMOVS4imZ2UwU7ngblJCYGLVlpxDIwbmFIBenYSZk4mztVYGaqUh11ekzGLdk9OZnZeqq2dBuzePvx2A36vBT0/BRUZuJ09+Zi91vHLwchAFfMHFaT58o4EbcQf56zjdzirtILKx9C1Ksz9BUTffgkMBkj+/lK2ST7KeaDTg11eQ/8//pb2neR8Qz0/DVUpfvDphpUWYKTaigz0wiziOIKem4K9vwD78yskf3sm1d5eC+j15YOtk5sA2w34n19BvVmH0gq+25fxiBO9K1XIQy/NwTy6dzAegUqJWyKPAAZWlhkDVSpCL84h+uNDmUTf2YPtD2QTvBN3Df1OE7b/So57jyL4JEHUHwDzU7L7Qbr18gebuAqymV6lBFPMQ0+Ow9yYRjKTVmyVEtzKhgRUqyNbHnsAvQH8Rh1e78rjnDxbUCsgiqCmxmG+vIPoD1/A3Lkh6xTzMcNqBDCwMupg8j2Wuazo4W2Zbt/chR8kcKtbQKtz/IPS5TP21aosYt7ehd/ehfndPZh7C9CT4+cfEX/Gc8OkB2NUy9CRQZTPQU+MwdxfQvKPX2F/eYXkxVv4zfrhDqLWAudt8BBFcvdzaR7xHx4i/tND6BszUPlYDoilzGNgZZ3WUOWi9Hse3ka00wCSBEli4dbkUIpjlVZi4esNJJ2O7JDQ7SHab8sdvMU52SCvVJSFxnEkldSHwiIyUm3lYujxClS1IuHU7sBt7MDWG6d3lThLHEFPVqGXZmHuL8Lcvgk1XvlsYwx09RhYGaeUkkqnmIdZmgPs1xIw/QSJ9zKVfnStISDh0e7Brm3BWwu/vgP7dBnmzk0JiqV5qLlJ2RWiVPjgMpjha/DDHpNSUqkNj+PSBlAfPtxV5SKoyXHouSmo2phspxMZnjs4QhhYo8IY6FoVKooO9kYHgETrw6U09viuDmh14Hp9+K1dqNfvYF+tIlrbhru3Bb04Bz0zIWsBi3kobSQY1dHGt5eGvpMTp32nB9/Yh3u3CftmXebDumdsxHceD9kep9eH227Arm/L7gylwsGl6rUY3qVPhoF1ia7zpLDcwDNApQhzd0H+bKwMFPOwP72Ee7sBv98+/kEuXQeYWFnE3OvD77WQPF+GrlZkSczEWDq2UAAKORltMGnVk26VjG5fLi+bLbjdffh6E67ekKb7TkPuAvoPn97ju7JUJ9FKHnO7gej3D6Dv3JTDJ847j/ESPHnyBI8fPz7+ej7i+33ZQXqVz32VGFijRCFdHD0uh43mcgfHgyVxBPduSyqv3kDuzg0Ng6vehG22gdfv5JCLQl56WmNlmbcqFo7NbXln4XsD+HYHvtECmi24Zhvo9GSPduuk2X70uZSSu4HDhdCJO/z3vQHcxrYcUdbYl5mu/gBRvw99YwZ6YgwoFmR9IQdIM4mBNSrSv2UVIOMJlRL07XnEkZZG9twUkp9+lcnyzbrMSB1txh/cwZPLRg8A7a4E3HZD7tTFsTTY07Dww7WDg0Qqol7/dJP/5GuMNJDLyWGokE0BMdwV1XtgYOH3WnDWIekP5M7n23VEX9+XmaylOdkgUPNHO4v4XR1VkYEeNq7HK1CT1bSRXZD92Ot7cv5ftyeXbM6d3qLGOmnY9/rw+wr+YHj0SA9ruMWMT/tZJy9ltAZiI5VZqSAVW7Uie7JbB7e2nW4E2Dk8eCKx8M02bH8At9uE29g52O/dt9O7mRPyuSGKZIg2Q5dFo4yBNWqOVlpRBJS0bO0SR1DVCszCLOzLVdnb6vU7uLfrssvnyUHTIe8B+xt7d0YDuRhqYkzGLpbmYO4uwNyYhqqU4FsdJE9fw/78CvbXFan8hlPvw7WIgwQ2sfD9AezmLqLXqzAPb8M8ugNz64YEIIdKM4OBNeqUkv3cZyehxivwC7OyW+niLJL5Kdipceltpbs9+EEi6wATC2+dNMvdkepp2G8aji8Me1JaluvAGLlsjCOgmIcaK8m6w7sLcljFozswN2eAYh6+0YKeqWFQLQO5GMhFsmtp58hR9YmFb7ZgOz247Qbc2hairV05V6OQh4kjuQTmXGkmMLBG3MGMlFYSKPkcTC6GqpSgF+fgvrkvW8Bs1eXgip10q5pmW5rf/YEsYO4nwCD9X+/lsdI3lYulIV8uSnN+vAw9UZWZqpkJ6Jma3AiYrkm1N16GiiL4clEeo1qGnq4h+WkW9sfnshZxv53eXUS6FrEPX7dwgwSJVlCzk9BzNahqGaZc+OiTeuh6YmDRIaVkTmu8LDsu3JwG+glcswW/tQu7sSP9oq1dOX15X84I9N2eHBeWjj7Ae2nAp9WNKuYlrIZBNTUOPTMJM1uDmq1BVytALpLnTqfnoeQka7MwI1XY9AT09AQG+RyQz8Gtbsg+W93+sQ0CfX8gf9btyd1O++GBVAoHA4sAHJnVUZA7bMZAIQfvPXS5AIyVoWZq8Lfm5VCJdtpsTy8R/cDKZVqSAPAyvW60jD/EsfSRCmljvVyUCq5SAtLN/g6f/0ivySjA5KAntPTY0srP3JxB8uNzJE9fH26dbGXBtpoYg7k5A3P7BvTSnCzd4fmEmcHAovdL57RgDEylCMzU5DQb5w77Vv7E3UDg8G6hQtrL0oe9LKPl7uDwfz/UEDeyFlHnYqhaFXp2EqiWgXIBtpCHW9mQu4ilAsy9BUTf3If56i7M0rxUb7wczAwGFp3t6N1EpY6tF/zc99uUVlKxaQ2Vk0vGONLQtTEkc9Owz5bh3qwBuQjxHx8i/vev5UCKsdJvP2uRrqVgvptZnqN5/Pgxvvvuu6t+GdefgqyJrJahCovQtSr01ATs3BTs82UAQPTtI5iv70PP1DK9A2mWfx/eJ7vfUcqcg1/SSBr0XsuYhJqoQN+eB7yX03OmJ6DyOR5GkUEMLApXLpJxiGoZZmkeAKS5n4sZVhnFwKJgKa2BnD67TzWil0xZx/lfIgoGKywKH6upkcEKi4iCwcAiomAwsIgoGAwsIgpG5pru1/kgCMqWj5k2v+yfy1H9OWeFRUTBYGARUTAYWEQUDAYWEQUjc033rHny5MmF3u9jtqe57Me86ON9isf8FNvynPXcf/7zny/9eejjKR/I7YaL3pEJ5NO5sE/xeV/2Y36Ku2VX+f3+FHtNZe3n8qrwkpCIgsHAIqJgMLCIKBgj3XS/yn2x2dN4v0/RP7tsofQNLyqEn0lWWEQUDAYWEQWDgUVEwWBgEVEwGFhEFAwGFhEFg4FFRMFgYBFRMBhYRBQMBhYRBWOkl+aMqqtcgnGVS2mu0lV+zUNYcnNRrLCIKBgMLCIKBgOLiILBwCKiYDCwiCgYDHUNnnQAAAGlSURBVCwiCgYDi4iCwcAiomAwsIgoGJx0H0GXPW1+lYcxfIxRPUA2S4cQs8IiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiILBwCKiYIz0pHsIk72fwqjuL37Zz32V0+afYso+BKywiCgYDCwiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKRuaW5lz2AQujLIQDIz7GZb/OUH7WeAgFEdEVYGARUTAYWEQUDAYWEQWDgUVEwWBgEVEwGFhEFAwGFhEFg4FFRMFQPoTxViIisMIiooAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiILBwCKiYDCwiCgYDCwiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiILBwCKiYDCwiCgYDCwiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgsHAIqJgMLCIKBgMLCIKBgOLiILBwCKiYDCwiCgYDCwiCgYDi4iCwcAiomAwsIgoGAwsIgoGA4uIgvH/AQMoBA/6AgFTAAAAAElFTkSuQmCC)
test1

---

# page2

-   item1
-   item2
"
`;
